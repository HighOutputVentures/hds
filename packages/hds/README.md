# HDS

We aim to build a library of custom ReactJS components that implements our unique UI design conventions (HDS). Our components will be based primarily on Chakra UI. React Storybook will be used for documentation and testing.

- Reusable block of components
- HDS integrated styling
- Easy to implement

## Installation

```bash
npm i @highoutput/hds
```

In your main app file import theme provider and make it a parent of all components.

```tsx
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ThemeProvider } from '@highoutput/hds';

const App = () => {
  return (<ThemeProvider>{...components here}</ThemeProvider>);
};

ReactDOM.render(<App />, document.getElementById('root'));
```

After that you can use all the components within hds core.

```jsx
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { InputField, Box } from '@highoutput/hds';

const Form = () => {
  return (
    <Box>
      <InputField
        helperMsg="This is a hint text to help user"
        id="name"
        label="Name"
        onBlur={() => {}}
        onChange={() => {}}
        onPressEnter={() => {}}
        placeholder="Input your name"
      />
    </Box>
  );
};

export default Form;
```

## Check out for more information

- [Storybook Documentation and Preview](https://hds-highoutput.vercel.app/?path=/story/getting-started--page)
- [Chakra Components](https://chakra-ui.com/docs/components)

## Other Packages

- [@highoutput/hds-icons](https://www.npmjs.com/package/@highoutput/hds-icons)
- [@highoutput/hds-auth](https://www.npmjs.com/package/@highoutput/hds-auth)
- [@highoutput/hds-spinner](https://www.npmjs.com/package/@highoutput/hds-spinner)
- [@highoutput/hds-pagination](https://www.npmjs.com/package/@highoutput/hds-pagination)
- [@highoutput/hds-array-field](https://www.npmjs.com/package/@highoutput/hds-array-field)
- [@highoutput/hds-auto-form](https://www.npmjs.com/package/@highoutput/hds-auto-form)
- [@highoutput/hds-contact-us](https://www.npmjs.com/package/@highoutput/hds-contact-us)
- [@highoutput/hds-utilities](https://www.npmjs.com/package/@highoutput/hds-utilities)
- [@highoutput/hds-forms](https://www.npmjs.com/package/@highoutput/hds-forms)
- [@highoutput/hds-autocomplete-input](https://www.npmjs.com/package/@highoutput/hds-autocomplete-input)
