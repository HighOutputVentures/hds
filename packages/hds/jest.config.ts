import { Config } from 'jest';

const config: Config = {
  displayName: 'hds',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nrwl/react/plugins/jest',
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/hds',
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
};

export default config;
