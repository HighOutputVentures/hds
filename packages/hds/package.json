{
  "version": "0.1.14",
  "license": "MIT",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "start": "tsdx watch",
    "build": "tsdx build",
    "test": "tsdx test --passWithNoTests",
    "lint": "tsdx lint",
    "prepare": "tsdx build",
    "size": "size-limit",
    "analyze": "size-limit --why",
    "generate:types": "chakra-cli tokens ./src/theme/index.ts"
  },
  "peerDependencies": {
    "react": ">=16"
  },
  "husky": {
    "hooks": {
      "pre-commit": "tsdx lint"
    }
  },
  "prettier": {
    "printWidth": 80,
    "semi": true,
    "singleQuote": true,
    "trailingComma": "es5"
  },
  "name": "@highoutput/hds",
  "author": {
    "name": "Niel Berongoy"
  },
  "module": "dist/hds.esm.js",
  "size-limit": [
    {
      "path": "dist/hds.cjs.production.min.js",
      "limit": "10 KB"
    },
    {
      "path": "dist/hds.esm.js",
      "limit": "10 KB"
    }
  ],
  "devDependencies": {
    "@size-limit/preset-small-lib": "7.0.8",
    "size-limit": "7.0.8"
  },
  "jest": {
    "moduleNameMapper": {
      "\\.(css|less|sass|scss)$": "<rootDir>/__mocks__/styleMock.js",
      "\\.(gif|ttf|eot|svg)$": "<rootDir>/__mocks__/fileMock.js"
    }
  },
  "gitHead": "220c468183d0032ba2f8eb5b26d89cf5712292c0",
  "dependencies": {
    "@chakra-ui/icons": "^2.0.2",
    "@chakra-ui/react": "^2.2.1",
    "@highoutput/hds-icons": "^0.1.5",
    "framer-motion": "^7.6.4",
    "lodash": "^4.17.21",
    "lodash-es": "^4.17.21"
  }
}
