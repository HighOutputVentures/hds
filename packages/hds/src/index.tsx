export {
  AspectRatio,
  AspectRatioProps,
  Box,
  BoxProps,
  Center,
  CenterProps,
  chakra as hds,
  Container,
  ContainerProps,
  Divider,
  DividerProps,
  Flex,
  FlexProps,
  forwardRef,
  Grid,
  GridItem,
  GridItemProps,
  GridProps,
  Heading,
  HeadingProps,
  HStack,
  Icon,
  IconProps,
  Image,
  ImageProps,
  Link,
  LinkBox,
  LinkBoxProps,
  LinkOverlay,
  LinkOverlayProps,
  LinkProps,
  List,
  ListIcon,
  ListItem,
  ListItemProps,
  ListProps,
  OrderedList,
  Portal,
  PortalProps,
  SimpleGrid,
  SimpleGridProps,
  Skeleton,
  SkeletonCircle,
  SkeletonProps,
  SkeletonText,
  SkeletonTextProps,
  Spacer,
  SpacerProps,
  Spinner,
  SpinnerProps,
  Stack,
  StackProps,
  Text,
  TextProps,
  UnorderedList,
  VisuallyHidden,
  VStack,
  Wrap,
  WrapItem,
  WrapItemProps,
  WrapProps,
} from '@chakra-ui/react';
export { default as ThemeProvider, ThemeProviderProps } from './components/ThemeProvider';
export { default as theme } from './theme';
export { default as extendTheme } from './utils/extendTheme';
