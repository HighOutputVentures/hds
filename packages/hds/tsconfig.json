{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "baseUrl": ".",
  },
  "include": ["src"],
  "exclude": ["node_modules", "types/*.ts"],
}
