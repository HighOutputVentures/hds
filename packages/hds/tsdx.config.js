const css = require('rollup-plugin-import-css');
const url = require('rollup-plugin-url');

module.exports = {
  rollup(config, options) {
    config.plugins = [
      ...config.plugins,
      css(),
      url({
        include: ['**/*.ttf', '**/*.css'],
        limit: Infinity,
      }),
    ];

    return config;
  },
};
