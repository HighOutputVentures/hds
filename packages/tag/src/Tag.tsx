import {
  Avatar,
  Box,
  SystemStyleObject,
  Tag,
  TagCloseButton,
  TagLabel,
  TagLeftIcon,
  Text,
  useBreakpoint,
  useMultiStyleConfig,
} from '@chakra-ui/react';
import * as React from 'react';
import Checkbox from './Checkbox';
import { omit, space } from './utils';

type Closable =
  | {
      closable?: false;
    }
  | {
      closable: true;
      onClose(): void;
    };

type Badgeable =
  | {
      badge?: false;
    }
  | {
      badge: true;
      badgeCount: number;
    };

type Checkable =
  | {
      checkbox?: false;
    }
  | {
      checkbox: true;
      checked?: boolean;
      onCheck(value: boolean): void;
    };

type Size = 'sm' | 'md' | 'lg';
type Breakpoints = 'base' | 'sm' | 'md' | 'lg' | 'xl';

type BaseProps = {
  label: string;
  size?: Size | Partial<Record<Breakpoints, Size>>;
  icon?: (props: React.ComponentProps<'svg'>) => JSX.Element;
  avatar?: string;
  indicator?: boolean;
  __testId?: string;
};

export type TagProps =
  SystemStyleObject /* We're using this as it doesn't come with props unrelated to styles üëç */ &
    BaseProps &
    Badgeable &
    Closable &
    Checkable;

/*
 *
 * HEADS UP! üì¢
 * We're heavily using Spacer due to design constraints. üòâ
 *
 */

export default React.forwardRef<HTMLDivElement, TagProps>(function HdsTag(p, ref) {
  const { variant, label, __testId, ...props } = Object.assign({ size: 'md' }, p, {
    variant: 'hds',
  });

  const size = useActualSize(props.size);
  const styles = useMultiStyleConfig('Tag', { size, variant });

  return (
    <Tag
      ref={ref}
      size={size}
      variant={variant}
      __css={{
        ...omit(
          props,
          'size',
          'icon',
          'badge',
          'avatar',
          'closable',
          'checkbox',
          'indicator',
          /* @ts-ignore "TypeScript doesn't know how to handle union types just yet ü§ê" */
          'onClose',
          'checked',
          'onCheck',
          'badgeCount',
        ),
        /* inbuilt styles should not be overridable ü´† */
        ...styles.container,
      }}
      data-testid={__testId}
    >
      {!!props.checkbox /* <!-- Checkbox --> */ && (
        <React.Fragment>
          {space(true, { width: { sm: '5px', md: '4px', lg: '5px' }[size] })}

          <Checkbox
            size={size}
            variant={variant}
            checked={props.checked}
            onChange={({ target }) => {
              props.onCheck(target.checked);
            }}
          />

          {space(![props.avatar, props.icon].some(Boolean), {
            width: { sm: '4px', md: '5px', lg: '6px' }[size],
          })}
        </React.Fragment>
      )}

      {!!props.icon /* <!-- Icon  --> */ && (
        <React.Fragment>
          {space(true, { width: { sm: '4px', md: '5px', lg: '6px' }[size] })}

          <TagLeftIcon as={props.icon} role="img" aria-label="Tag Icon" sx={styles.icon} />

          {space(true, { width: { sm: '4px', md: '5px', lg: '6px' }[size] })}
        </React.Fragment>
      )}

      {!!props.avatar /* <!-- Avatar --> */ && (
        <React.Fragment>
          {space(true, { width: { sm: '4px', md: '5px', lg: '6px' }[size] })}

          <Avatar src={props.avatar} role="img" aria-label="Tag Avatar" sx={styles.avatar} />

          {space(true, { width: { sm: '4px', md: '5px', lg: '6px' }[size] })}
        </React.Fragment>
      )}

      {!!props.indicator /* <!-- Indicator --> */ && (
        <React.Fragment>
          {space(true, { width: { sm: '7px', md: '8px', lg: '10px' }[size] })}

          <Box role="presentation" aria-label="Green Dot" sx={styles.indicator} />

          {space(true, { width: { sm: '5px', md: '6px', lg: '7px' }[size] })}
        </React.Fragment>
      )}

      {!!label /* <!-- Label --> */ && (
        <React.Fragment>
          {space(![props.icon, props.avatar, props.indicator, props.checkbox].some(Boolean), {
            width: { sm: '8px', md: '9px', lg: '10px' }[size],
          })}

          <TagLabel role="contentinfo" __css={styles.label}>
            {label}
          </TagLabel>

          {space(![props.closable, props.badge].some(Boolean), {
            width: { sm: '8px', md: '9px', lg: '10px' }[size],
          })}
        </React.Fragment>
      )}

      {!!props.closable /* <!-- Close Button --> */ && (
        <React.Fragment>
          {space(true, { width: { sm: '5px', md: '5px', lg: '6px' }[size] })}

          <TagCloseButton role="button" aria-label="Close Tag" onClick={props.onClose} />

          {space(true, { width: { sm: '7px', md: '6px', lg: '7px' }[size] })}
        </React.Fragment>
      )}

      {!!props.badge /* <!-- Badge --> */ && (
        <React.Fragment>
          {space(true, { width: { sm: '8px', md: '10px', lg: '12px' }[size] })}

          <Text sx={styles.badge}>{props.badgeCount}</Text>

          {space(true, { width: { sm: '8px', md: '8px', lg: '10px' }[size] })}
        </React.Fragment>
      )}
    </Tag>
  );
});

export function useActualSize(size: TagProps['size']) {
  const actualBreakpoint = useBreakpoint();
  const psuedoBreakpoint = React.useMemo(() => {
    return 'sm|md|lg'.split(/\|/g).includes(actualBreakpoint)
      ? actualBreakpoint
      : ['base'].includes(actualBreakpoint)
      ? 'sm'
      : 'lg';
  }, []);

  const keys = Object.keys(size);

  return typeof size === 'string'
    ? size
    : size[psuedoBreakpoint] ?? /* fallback to closest given size */ size[keys[keys.length - 1]];
}
