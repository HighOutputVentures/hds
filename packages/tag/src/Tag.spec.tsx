import { ThemeProvider } from '@highoutput/hds';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, screen, waitFor, within } from '@testing-library/react';
import * as React from 'react';
import renderer from 'react-test-renderer';
import AUIcon from './examples/AUIcon';
import Tag from './Tag';

const handleCheck = jest.fn();
const handleClose = jest.fn();

jest.mock('@chakra-ui/react', () => {
  return {
    ...jest.requireActual('@chakra-ui/react'),
    useBreakpoint() {
      return 'lg';
    },
  };
});

describe('Tag', () => {
  beforeEach(() => {
    render(
      <Tag
        label="Tag Label"
        indicator
        icon={AUIcon}
        avatar="https://i.pravatar.cc/25"
        checkbox
        checked
        onCheck={handleCheck}
        badge
        badgeCount={1_000_000}
        closable
        onClose={handleClose}
        __testId="tag"
      />,
    );
  });

  it('Should render tag', () => {
    expect(screen.queryByTestId('tag')).toBeDefined();
  });

  it('Should render correct label', () => {
    expect(screen.queryByRole('contentinfo')).toBeDefined();
    expect(screen.getByRole('contentinfo')).toHaveTextContent('Tag Label');
  });

  it('Should render indicator', () => {
    expect(screen.queryByRole('presentation', { name: 'Green Dot' })).toBeDefined();
  });

  it('Should render icon', () => {
    expect(screen.queryByRole('img', { name: 'Tag Icon' })).toBeDefined();
  });

  it('Should render avatar', () => {
    expect(screen.queryByRole('img', { name: 'Tag Avatar' })).toBeDefined();
  });

  it('Should render badge', () => {
    expect(screen.queryByText('1000000')).toBeInTheDocument();
  });

  it('Should render close button', () => {
    expect(screen.queryByRole('button', { name: 'Close Tag' })).toBeDefined();
  });

  it('Should call onClose callback when close button is clicked', async () => {
    fireEvent.click(screen.getByRole('button', { name: 'Close Tag' }));

    await waitFor(() => {
      expect(handleClose).toHaveBeenCalled();
    });
  });

  it('Should render checkbox', () => {
    expect(screen.queryByRole('checkbox', { name: 'Select Item' })).toBeDefined();
  });

  it('Should be able to toggle checkbox checked', async () => {
    const { getByTestId } = render(
      <Tag label="Isolated Tag Label" checkbox onCheck={handleCheck} __testId="isolated-tag" />,
    );

    const checkbox = within(getByTestId('isolated-tag')).getByRole('checkbox', {
      name: 'Select Item',
    });

    fireEvent.click(checkbox);

    await waitFor(() => {
      expect(handleCheck).toHaveBeenCalledWith(true);
    });
  });

  it('Should call checkbox onChange callback on toggle', async () => {
    fireEvent.click(screen.getByRole('checkbox', { name: 'Select Item' }));

    await waitFor(() => {
      expect(handleCheck).toHaveBeenCalledWith(expect.any(Boolean));
    });
  });

  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const component = renderer.create(
        <Tag
          label="Tag Label"
          indicator
          icon={AUIcon}
          avatar="https://i.pravatar.cc/25"
          checkbox
          checked
          badge
          badgeCount={1_000_000}
          closable
          onCheck={function () {}}
          onClose={function () {}}
        />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });

  it('Should match snapshot', () => {
    const component = renderer.create(
      <ThemeProvider>
        <Tag
          label="Tag Label"
          indicator
          icon={AUIcon}
          avatar="https://i.pravatar.cc/25"
          checkbox
          checked
          onCheck={function () {}}
          badge
          badgeCount={1_000_000}
          closable
          onClose={function () {}}
        />
      </ThemeProvider>,
    );

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
    component.unmount();
  });
});
