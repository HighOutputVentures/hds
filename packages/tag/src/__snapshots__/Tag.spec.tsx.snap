// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Should match snapshot 1`] = `
Array [
  <span
    className="css-1ipiso8"
    data-testid="hds.tag"
  >
    <div
      className="css-1w2j5n9"
    />
    <label
      checked={true}
      className="chakra-checkbox css-1577qb8"
      data-testid="hds.tag.checkbox"
      onClick={[Function]}
    >
      <input
        aria-label="Select Item"
        checked={false}
        className="chakra-checkbox__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "border": "0",
            "clip": "rect(0, 0, 0, 0)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": "0",
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "1px",
          }
        }
        type="checkbox"
      />
      <span
        aria-hidden={true}
        className="chakra-checkbox__control css-1ghg39o"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
      />
    </label>
    <div
      className="css-sfr4vn"
    />
    <svg
      aria-label="Tag Icon"
      className="chakra-icon css-1jxv7ty"
      data-testid="hds.tag.left.icon"
      fill="none"
      focusable={false}
      height="16"
      role="img"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clipPath="url(#clip0_3307_418389)"
      >
        <path
          d="M16 8C16 12.4183 12.4183 16 8 16C3.58175 16 0 12.4183 0 8C0 8.00187 8 0.00090625 8 0C12.4183 0 16 3.58175 16 8Z"
          fill="#0052B4"
        />
        <path
          d="M7.97852 8.00026H7.99996C7.99996 7.993 7.99996 7.98605 7.99996 7.97882C7.99283 7.98599 7.98569 7.99313 7.97852 8.00026Z"
          fill="#F0F0F0"
        />
        <path
          d="M7.99987 4.17442C7.99987 2.76677 7.99987 1.84436 7.99987 0.000518799H7.99853C3.58088 0.00123755 -0.00012207 3.58264 -0.00012207 8.00049H4.17378V5.65011L6.52416 8.00049H7.9785C7.98565 7.99336 7.99278 7.98624 7.9999 7.97908C7.9999 7.44011 7.9999 6.95921 7.9999 6.52483L5.6495 4.17442H7.99987Z"
          fill="#F0F0F0"
        />
        <path
          d="M4.04745 1.04379C2.79624 1.75626 1.75605 2.79645 1.04358 4.04767V8.00033H3.13055V3.13079V3.13076H8.00011C8.00011 2.47254 8.00011 1.84548 8.00011 1.04379H4.04745Z"
          fill="#D80027"
        />
        <path
          d="M8.00025 7.01658L5.15797 4.17433H4.17419C4.17419 4.1743 4.17419 4.17433 4.17419 4.17433L8.00022 8.00036H8.00025C8.00025 8.00036 8.00025 7.32208 8.00025 7.01658Z"
          fill="#D80027"
        />
        <path
          d="M4.82498 9.39096L5.26404 10.309L6.25551 10.0799L5.81151 10.9956L6.60883 11.6278L5.61611 11.8516L5.61889 12.8692L4.82498 12.2326L4.03111 12.8692L4.03389 11.8516L3.04114 11.6278L3.83848 10.9956L3.39442 10.0799L4.38595 10.309L4.82498 9.39096Z"
          fill="#F0F0F0"
        />
        <path
          d="M11.9781 11.1302L12.1976 11.5893L12.6934 11.4747L12.4713 11.9325L12.87 12.2487L12.3736 12.3606L12.375 12.8694L11.9781 12.551L11.5811 12.8694L11.5825 12.3606L11.0862 12.2487L11.4848 11.9325L11.2628 11.4747L11.7586 11.5893L11.9781 11.1302Z"
          fill="#F0F0F0"
        />
        <path
          d="M9.93564 6.26096L10.1551 6.72002L10.6509 6.60539L10.4289 7.06324L10.8275 7.37939L10.3312 7.49127L10.3325 8.00008L9.93564 7.68174L9.5387 8.00008L9.54004 7.49127L9.0437 7.37939L9.44236 7.06324L9.22036 6.60539L9.71611 6.72002L9.93564 6.26096Z"
          fill="#F0F0F0"
        />
        <path
          d="M11.9781 3.47842L12.1976 3.93749L12.6934 3.82289L12.4714 4.28074L12.87 4.59686L12.3736 4.70877L12.375 5.21758L11.9781 4.89924L11.5811 5.21758L11.5825 4.70877L11.0862 4.59686L11.4848 4.28074L11.2628 3.82289L11.7586 3.93749L11.9781 3.47842Z"
          fill="#F0F0F0"
        />
        <path
          d="M13.7614 5.56519L13.9809 6.02425L14.4767 5.90962L14.2547 6.36747L14.6533 6.68362L14.157 6.79553L14.1583 7.30431L13.7614 6.986L13.3645 7.30431L13.3659 6.79553L12.8695 6.68362L13.2682 6.36747L13.0462 5.90962L13.5419 6.02425L13.7614 5.56519Z"
          fill="#F0F0F0"
        />
        <path
          d="M12.4861 8.00021L12.6588 8.53165H13.2176L12.7655 8.86012L12.9382 9.39153L12.4861 9.06309L12.034 9.39153L12.2067 8.86012L11.7546 8.53165H12.3134L12.4861 8.00021Z"
          fill="#F0F0F0"
        />
      </g>
      <defs>
        <clipPath
          id="clip0_3307_418389"
        >
          <rect
            fill="white"
            height="16"
            width="16"
          />
        </clipPath>
      </defs>
    </svg>
    <div
      className="css-sfr4vn"
    />
    <div
      className="css-sfr4vn"
    />
    <span
      aria-label="Tag Avatar"
      className="chakra-avatar css-sb6ugy"
      data-testid="hds.tag.avatar"
      role="img"
    >
      <svg
        aria-label=" avatar"
        className="chakra-avatar__svg css-16ite8i"
        role="img"
        viewBox="0 0 128 128"
      >
        <path
          d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
          fill="currentColor"
        />
        <path
          d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      className="css-sfr4vn"
    />
    <div
      className="css-zgv5qp"
    />
    <div
      aria-label="Green Dot"
      className="css-0"
      data-testid="hds.tag.indicator"
      role="presentation"
    />
    <div
      className="css-xf646l"
    />
    <span
      className="css-1ny2kle"
      data-testid="hds.tag.label"
      role="contentinfo"
    >
      Tag Label
    </span>
    <div
      className="css-sfr4vn"
    />
    <button
      aria-label="Close Tag"
      className="css-e9kjw6"
      data-testid="hds.tag.close.button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      <svg
        className="chakra-icon css-1md6f3y"
        focusable={false}
        viewBox="0 0 512 512"
      >
        <path
          d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      className="css-xf646l"
    />
    <div
      className="css-fxo5v0"
    />
    <p
      className="chakra-text css-0"
    >
      1000000
    </p>
    <div
      className="css-zgv5qp"
    />
  </span>,
  <span />,
]
`;

exports[`Tag Snapshot Should match snapshot 1`] = `
<span
  className="css-1qm7lvz"
  data-testid="hds.tag"
>
  <div
    className="css-1w2j5n9"
  />
  <label
    checked={true}
    className="chakra-checkbox css-1elxtiq"
    data-testid="hds.tag.checkbox"
    onClick={[Function]}
  >
    <input
      aria-label="Select Item"
      checked={false}
      className="chakra-checkbox__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        Object {
          "border": "0",
          "clip": "rect(0, 0, 0, 0)",
          "height": "1px",
          "margin": "-1px",
          "overflow": "hidden",
          "padding": "0",
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "1px",
        }
      }
      type="checkbox"
    />
    <span
      aria-hidden={true}
      className="chakra-checkbox__control css-dnty2r"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
    />
  </label>
  <div
    className="css-sfr4vn"
  />
  <svg
    aria-label="Tag Icon"
    className="chakra-icon css-1jxv7ty"
    data-testid="hds.tag.left.icon"
    fill="none"
    focusable={false}
    height="16"
    role="img"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      clipPath="url(#clip0_3307_418389)"
    >
      <path
        d="M16 8C16 12.4183 12.4183 16 8 16C3.58175 16 0 12.4183 0 8C0 8.00187 8 0.00090625 8 0C12.4183 0 16 3.58175 16 8Z"
        fill="#0052B4"
      />
      <path
        d="M7.97852 8.00026H7.99996C7.99996 7.993 7.99996 7.98605 7.99996 7.97882C7.99283 7.98599 7.98569 7.99313 7.97852 8.00026Z"
        fill="#F0F0F0"
      />
      <path
        d="M7.99987 4.17442C7.99987 2.76677 7.99987 1.84436 7.99987 0.000518799H7.99853C3.58088 0.00123755 -0.00012207 3.58264 -0.00012207 8.00049H4.17378V5.65011L6.52416 8.00049H7.9785C7.98565 7.99336 7.99278 7.98624 7.9999 7.97908C7.9999 7.44011 7.9999 6.95921 7.9999 6.52483L5.6495 4.17442H7.99987Z"
        fill="#F0F0F0"
      />
      <path
        d="M4.04745 1.04379C2.79624 1.75626 1.75605 2.79645 1.04358 4.04767V8.00033H3.13055V3.13079V3.13076H8.00011C8.00011 2.47254 8.00011 1.84548 8.00011 1.04379H4.04745Z"
        fill="#D80027"
      />
      <path
        d="M8.00025 7.01658L5.15797 4.17433H4.17419C4.17419 4.1743 4.17419 4.17433 4.17419 4.17433L8.00022 8.00036H8.00025C8.00025 8.00036 8.00025 7.32208 8.00025 7.01658Z"
        fill="#D80027"
      />
      <path
        d="M4.82498 9.39096L5.26404 10.309L6.25551 10.0799L5.81151 10.9956L6.60883 11.6278L5.61611 11.8516L5.61889 12.8692L4.82498 12.2326L4.03111 12.8692L4.03389 11.8516L3.04114 11.6278L3.83848 10.9956L3.39442 10.0799L4.38595 10.309L4.82498 9.39096Z"
        fill="#F0F0F0"
      />
      <path
        d="M11.9781 11.1302L12.1976 11.5893L12.6934 11.4747L12.4713 11.9325L12.87 12.2487L12.3736 12.3606L12.375 12.8694L11.9781 12.551L11.5811 12.8694L11.5825 12.3606L11.0862 12.2487L11.4848 11.9325L11.2628 11.4747L11.7586 11.5893L11.9781 11.1302Z"
        fill="#F0F0F0"
      />
      <path
        d="M9.93564 6.26096L10.1551 6.72002L10.6509 6.60539L10.4289 7.06324L10.8275 7.37939L10.3312 7.49127L10.3325 8.00008L9.93564 7.68174L9.5387 8.00008L9.54004 7.49127L9.0437 7.37939L9.44236 7.06324L9.22036 6.60539L9.71611 6.72002L9.93564 6.26096Z"
        fill="#F0F0F0"
      />
      <path
        d="M11.9781 3.47842L12.1976 3.93749L12.6934 3.82289L12.4714 4.28074L12.87 4.59686L12.3736 4.70877L12.375 5.21758L11.9781 4.89924L11.5811 5.21758L11.5825 4.70877L11.0862 4.59686L11.4848 4.28074L11.2628 3.82289L11.7586 3.93749L11.9781 3.47842Z"
        fill="#F0F0F0"
      />
      <path
        d="M13.7614 5.56519L13.9809 6.02425L14.4767 5.90962L14.2547 6.36747L14.6533 6.68362L14.157 6.79553L14.1583 7.30431L13.7614 6.986L13.3645 7.30431L13.3659 6.79553L12.8695 6.68362L13.2682 6.36747L13.0462 5.90962L13.5419 6.02425L13.7614 5.56519Z"
        fill="#F0F0F0"
      />
      <path
        d="M12.4861 8.00021L12.6588 8.53165H13.2176L12.7655 8.86012L12.9382 9.39153L12.4861 9.06309L12.034 9.39153L12.2067 8.86012L11.7546 8.53165H12.3134L12.4861 8.00021Z"
        fill="#F0F0F0"
      />
    </g>
    <defs>
      <clipPath
        id="clip0_3307_418389"
      >
        <rect
          fill="white"
          height="16"
          width="16"
        />
      </clipPath>
    </defs>
  </svg>
  <div
    className="css-sfr4vn"
  />
  <div
    className="css-sfr4vn"
  />
  <span
    aria-label="Tag Avatar"
    className="chakra-avatar css-1ruhx4w"
    data-testid="hds.tag.avatar"
    role="img"
  >
    <svg
      aria-label=" avatar"
      className="chakra-avatar__svg css-16ite8i"
      role="img"
      viewBox="0 0 128 128"
    >
      <path
        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
        fill="currentColor"
      />
      <path
        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
        fill="currentColor"
      />
    </svg>
  </span>
  <div
    className="css-sfr4vn"
  />
  <div
    className="css-zgv5qp"
  />
  <div
    aria-label="Green Dot"
    className="css-0"
    data-testid="hds.tag.indicator"
    role="presentation"
  />
  <div
    className="css-xf646l"
  />
  <span
    className="css-zvlevn"
    data-testid="hds.tag.label"
    role="contentinfo"
  >
    Tag Label
  </span>
  <div
    className="css-sfr4vn"
  />
  <button
    aria-label="Close Tag"
    className="css-sorhh"
    data-testid="hds.tag.close.button"
    onClick={[Function]}
    role="button"
    type="button"
  >
    <svg
      className="chakra-icon css-1md6f3y"
      focusable={false}
      viewBox="0 0 512 512"
    >
      <path
        d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    className="css-xf646l"
  />
  <div
    className="css-fxo5v0"
  />
  <p
    className="chakra-text css-0"
  >
    1000000
  </p>
  <div
    className="css-zgv5qp"
  />
</span>
`;
